app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java

```diff
@@ -172,7 +172,7 @@ private boolean isCredentialsValid(final String username, final String password)
 
         final Resources resources = context.getResources();
         final String correctUsername = username.replaceFirst("\\s++$", "").trim();
-        if (username == null || username.trim().isEmpty()) {
+        if (username == null || username.matches("\\s*") || username.isEmpty()) {
             getMvpView().showUsernameError(context.getString(R.string.error_validation_blank,
```

Comment:
PMD added [InefficientEmptyStringCheck](https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#InefficientEmptyStringCheck) in PMD 3.6 which was updated with the gradle plugin update.